astroplan/telescope.py

```diff
@@ -0,0 +1,139 @@
+				93.52139537, 92.57268607, 91.77633908, 90.30410094,
+				88.68846298, 86.74856762, 84.85909033, 82.85400237,
+				80.76562301, 78.18504085, 75.90628116, 72.63150731,
+				68.93418199, 65.34249453, 61.79608045, 58.13799205,
+				54.49429826, 49.87975185, 45.18326838, 41.57397462,
+				36.82027063, 32.80603172, 28.7917928, 25.09446748,
+				21.39714216, 18.4392819, 14.89286782, 0.]
+	return np.array(wl) * u.nm, np.array(response) * u.Unit('') / 100
+
+
+class Telescope(object):
+	"""
+	A class to store telescope specifications.
+	"""
+	@u.quantity_input(diameter=u.m, gain=u.ct / u.adu,
+					  readnoise=u.ct / u.pixel, ad_err=u.adu / u.pixel)
+	def __init__(self, diameter, bandpass, gain, ccd_response=False,
+		readnoise=0 * (u.ct / u.pixel), ad_err=AD_ERR_DEFAULT):
+		"""
+		Parameters
+		----------
+		diameter : `~astropy.units.Quantity`
+			Diameter of the telescope aperture.
+		bandpass : `~synphot.spectrum.SpectralElement` or 2D-array
+			The bandpass of the instrument/telescope. Can either be
+			specified by a `~synphot.spectrum.SpectralElement` object
+			or an array with dimensions 2 x n, where one row is the
+			array of wavelengths associated with the bandpass, and the other
+			row contains the fractional values of the bandpass.
+		gain : `~astropy.units.Quantity`, optional
+			Gain of the CCD with units of counts/ADU. Default is
+			1 * astropy.units.ct / astropy.units.adu such that the
+			contribution to the error due to the gain is assumed to be small.
+		ccd_response : 'default' or 2D-array
+			Note: the zeroth index of the given array must be the
+			waveset of the response function, while the other
+			must contain the values of the function.
+		readnoise : `~astropy.units.Quantity`, optional
+			Counts per pixel from the read noise with units of counts/pixel.
+			Default is 0 * astropy.units.ct / astropy.units.pixel.
+		ad_err : `~astropy.units.Quantity`, optional
+			An estimate of the 1 sigma error within the A/D converter with units of
+			adu/pixel. Default is set to
+			sqrt(0.289) * astropy.units.adu / astropy.units.pixel, where
+			sqrt(0.289) comes from the assumption that "for a charge level that is
+			half way between two output ADU steps, there is an equal chance that it
+			will be assigned to the lower or to the higher ADU value when converted
+			to a digital number" (text from footnote on page 56 of [1]_, see also
+			[2]_).
+
+			References
+	   		----------
+	    .. [1] Howell, S. B. 2000, *Handbook of CCD Astronomy* (Cambridge, UK:
+	        Cambridge University Press)
+	    .. [2] Merline, W. & Howell, S. B. *A Realistic Model for Point-sources
+	        Imaged on Array Detectors: The Model and Initial Results*. ExA, 6:163
+	        (1995)
+		"""
+		self.diameter = diameter
+		self.gain = gain
+		self.ad_err = ad_err
+		self.readnoise = readnoise
+		self._bandpass = bandpass
+		self._ccd_response = ccd_response
+
+	@property
+	def area(self):
+		""" Area of the telescope. """
```

Comment:
I think things are overindented here, 4 spaces is the typical one. Same comment applies to other methods in this file.